tests/v2/validation/rancherha/main.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"github.com/rancher/rancher/tests/framework/clients/corral"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/pipelineutils"
+	"github.com/rancher/rancher/tests/framework/pkg/environmentflag"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	corralPackageRegistryRancherName = "registryrancher"
+	corralPackageRancherName         = "rancherha"
+)
+
+func main() {
+	corralSession := session.NewSession()
+
+	corralConfig := corral.CorralConfigurations()
+	err := corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	if err != nil {
+		logrus.Fatalf("error setting up corral: %v", err)
+	}
+
+	configPackage := corral.CorralPackagesConfig()
+
+	environmentFlags := environmentflag.NewEnvironmentFlags()
+	environmentflag.LoadEnvironmentFlags(environmentflag.ConfigurationFileKey, environmentFlags)
+	installRancher := environmentFlags.GetValue(environmentflag.InstallRancher)
+	rancherWithRegistries := environmentFlags.GetValue(environmentflag.InstallRancherWithRegistry)
+
+	logrus.Infof("installRancher value is %t", installRancher)
+	logrus.Infof("rancherWithRegistries value is %t", rancherWithRegistries)
+
+	if installRancher {
+		path := configPackage.CorralPackageImages[corralPackageRancherName]
+		corralName := corralPackageRancherName
+		if rancherWithRegistries {
```

Comment:
Is there a specific reason we have different flag for the rancher with registries?

Reply:
There is, using this logic we can opt for either a Rancher HA install without global registry or with it.
We can also create more flags like Rancher with Proxy. I thought on making it generic as possible.

Reply:
How would you recommend to use this logic of using only `rancherha` in the test suite.
I use these flags to skip some tests depending on the Rancher setup has registry enabled or not.

Reply:
I removed the flag and moved the logic to check a Corral variable during test setup